packages/toggle/src/ToggleStyle.tsx

```diff
@@ -0,0 +1,37 @@
+import React, { useMemo } from 'react';
+import { useEditorRef } from '@udecode/plate-common';
+import { TIndentElement } from '@udecode/plate-indent';
+
+import { idToClassName } from './injectNodeIdClassName';
+import { findElementIdsHiddenInToggle } from './queries';
+import { useToggleControllerStore } from './store';
+
+// TODO restrict to the editor
+export const ToggleStyle = () => {
+  const { children } = useEditorRef();
+  const [openIds] = useToggleControllerStore().use.openIds();
+  const css = useMemo(
+    () => hideElementsInToggleCSS(openIds, children as TIndentElement[]),
+    [openIds, children]
+  );
+
+  return <style>{css}</style>;
```

Comment:
Nice idea, but there are a couple of problems with this approach:

1. This won't work in apps that use a restrictive content security policy that blocks inline style elements (and for good reason—see point 2).
2. The code as it's currently written is vulnerable to CSS injection if a user opens a malicously crafted document, which can be used to install [CSS key loggers](https://css-tricks.com/css-keylogger/) on the page.

Reply:
A better solution might be to rely on derived Jotai atoms to ensure that components only re-render when a specific ID is modified.

Reply:
I see 👍 Any idea on how to derive a state from `editor.children`? It's the missing piece

Reply:
`useEditorSelector` is the most optimal one